### PR TITLE
History management

### PR DESCRIPTION
**Pull request by kshitijthatte**
- added a single video page ( functionality on it is yet to be updated)
- added services and created a history context 
- opening a video, adds it to the history
- history page can be viewed from the nav drawer, where
  - single video can be removed from history
  - full history can also be deleted

### CODE REVIEW

File: src/components/AppRoutes.js [hunk 1]
```diff
@@ -7,18 +7,22 @@ import Playlists from "../pages/Playlists";
 import SinglePlaylist from "../pages/SinglePlaylist";
 import LikedVideos from "../pages/LikedVideos";
 import WatchLater from "../pages/WatchLater";
+import SingleVideo from "../pages/SingleVideo";
```
Comment (mimansha-swarup) [inline src/components/AppRoutes.js:10]: you can use barrel export

File: src/pages/SinglePlaylist/HorizontalVideoCard.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { useAuth } from "../../contexts/authContext";
+import { useUserHistory } from "../../contexts/historyContext";
 import { useLikes } from "../../contexts/likesContext";
 import { usePlaylist } from "../../contexts/playlistContext";
 import { useWatchlater } from "../../contexts/watchlaterContext";
```
Comment (mimansha-swarup) [inline src/pages/SinglePlaylist/HorizontalVideoCard.js:5]: here too barrel exports can be used

**Review by mimansha-swarup — APPROVED** [review]
code is clean
redable
just try using barrel exports more

**Review by preethasri — APPROVED** [review]
UI looks neat and good .folder structure is good .neat work👍

File: src/pages/SinglePlaylist/HorizontalVideoCard.js (hunk 2 above)
Comment (kshitijthatte) [reply src/pages/SinglePlaylist/HorizontalVideoCard.js:5]: Sure, will update that.

**Pull request merged**